platform/jewel/ui/src/main/kotlin/org/jetbrains/jewel/ui/component/Tooltip.kt

```diff
@@ -154,7 +138,33 @@ public fun Tooltip(
 }
 
 @Composable
-private fun TooltipImpl(style: TooltipStyle, tooltip: @Composable () -> Unit) {
+private fun TooltipImpl(
+    enabled: Boolean,
+    style: TooltipStyle,
+    autoHideBehavior: TooltipAutoHideBehavior = style.autoHideBehavior,
+    tooltip: @Composable () -> Unit,
+) {
+    var isVisible by remember { mutableStateOf(true) }
+    if (!enabled) return
+
+    LaunchedEffect(style.autoHideBehavior) {
+        when (style.autoHideBehavior) {
+            TooltipAutoHideBehavior.Normal -> {
+                delay(style.metrics.regularDisappearDelay)
+                isVisible = false
+            }
+
+            TooltipAutoHideBehavior.Long -> {
+                delay(style.metrics.fullDisappearDelay)
+                isVisible = false
+            }
+
+            TooltipAutoHideBehavior.Never -> {}
+        }
+    }
+
+    if (!isVisible) return
```

Comment:
We had this comment for the `!isVisible` case:

```kotlin
// Render an empty box when not visible to allow TooltipArea to dismiss properly
```

Are you sure removing the empty Box does not break anything?

Reply:
Ok, seems fine. If we find issues we can restore it :)